auraed/src/runtime/mod.rs

```diff
@@ -28,95 +28,85 @@
  *                                                                            *
 \* -------------------------------------------------------------------------- */
 
-mod cell_name;
-mod cgroup_table;
-mod child_table;
+mod cells_table;
 mod cpu_cpus;
 mod cpu_quota;
 mod error;
-mod executable_name;
 mod validation;
 
-use crate::runtime::cell_name::CellName;
-use crate::runtime::error::Result;
-use crate::runtime::executable_name::ExecutableName;
+pub(crate) use self::validation::ValidatedCell;
+use crate::cells::Cell;
+use crate::runtime::cells_table::CellsTable;
+use crate::runtime::error::RuntimeError;
 use crate::runtime::validation::{
-    ValidatedAllocateCellRequest, ValidatedCell, ValidatedExecutable,
+    ValidatedAllocateCellRequest, ValidatedExecutable,
     ValidatedFreeCellRequest, ValidatedStartCellRequest,
     ValidatedStopCellRequest,
 };
-use crate::runtime::{
-    cgroup_table::CgroupTable, child_table::ChildTable, error::RuntimeError,
-};
 use ::validation::ValidatedType;
 use anyhow::anyhow;
 use aurae_proto::runtime::{
     cell_service_server, AllocateCellRequest, AllocateCellResponse,
     FreeCellRequest, FreeCellResponse, StartCellRequest, StartCellResponse,
     StopCellRequest, StopCellResponse,
 };
-use cgroups_rs::{cgroup_builder::CgroupBuilder, *};
 use log::info;
-use std::{io, os::unix::process::CommandExt, process::Command};
 use tonic::{Request, Response, Status};
 
 #[derive(Debug, Clone)]
 pub struct CellService {
-    child_table: ChildTable,
-    cgroup_table: CgroupTable,
+    cells: CellsTable,
 }
 
 impl CellService {
     pub fn new() -> Self {
-        CellService {
-            child_table: Default::default(),
-            cgroup_table: Default::default(),
-        }
-    }
-
-    fn aurae_process_pre_exec(exe_name: &ExecutableName) -> io::Result<()> {
-        info!("CellService: aurae_process_pre_exec(): {exe_name}");
-        // Here we are executing as the new spawned pid.
-        // This is a place where we can "hook" into all processes
-        // started with Aurae in the future. Similar to kprobe/uprobe
-        // in Linux or LD_PRELOAD in libc.
-        Ok(())
+        CellService { cells: Default::default() }
     }
 
     fn allocate(
         &self,
         request: ValidatedAllocateCellRequest,
-    ) -> std::result::Result<Response<AllocateCellResponse>, Status> {
+    ) -> Result<Response<AllocateCellResponse>, Status> {
         // Initialize the cell
         let ValidatedAllocateCellRequest { cell } = request;
-
         info!("CellService: allocate() cell={:?}", cell);
 
+        if self.cells.contains(&cell.name).map_err(RuntimeError::from)? {
```

Comment:
this could be an error on cells.insert, no?

Reply:
Yes, but then you would have already called 'Cell::allocate` (ln 84)

Reply:
It checks on insert too, I think